### PR TITLE
chore(samples/backend/frontend): Update preload control-flow and data-passing pipelines to v2

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Part of #9535

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: samples/tutorials/DSL - Control structures/DSL - Control structures.py [hunk 1]
```diff
@@ -49,75 +41,63 @@ def flip_coin_op() -> str:
...
-                print_op('tails and %s > 15!' % random_num_tail.output)
-            with dsl.Condition(random_num_tail.output <= 15):
-                print_op('tails and %s <= 15!' % random_num_tail.output)
+                print_op(
+                    message=f'{expected_result} and {random_num_head.output} <= 5!'
+                )
 
-        with dsl.Condition(flip.output == 'tails'):
-            fail_op(message="Failing the run to demonstrate that exit handler still gets executed.")
+        # with dsl.Condition(expected_result == 'tails'):
```
Comment (Linchin) [inline samples/tutorials/DSL - Control structures/DSL - Control structures.py:97]: It looks like we need to add a situation where the task fails so we can run the exit handler.

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py [hunk 2]
```diff
@@ -16,149 +16,130 @@
...
+    # of type `Dataset`.
+    dataset_one_path: InputPath('Dataset'),
+    # Use Input[T] to get a metadata-rich handle to the input artifact
+    # of type `Dataset`.
+    dataset_two: Input[Dataset],
+    # An input parameter of type string.
+    message: str,
+    # Use Output[T] to get a metadata-rich handle to the output artifact
+    # of type `Dataset`.
+    model: Output[Model],
```
Comment (Linchin) [inline samples/tutorials/Data passing in python components/Data passing in python components - Files.py:98]: Should this be `model: Output[Dataset]`?

**Comment by Linchin** [comment]
Hi @chensun, thank you for updating the v2 samples! Overall it looks good, except for a few small issues. Also the frontend test is failing.

File: samples/tutorials/Data passing in python components/Data passing in python components - Files.py (hunk 2 above)
Comment (chensun) [reply samples/tutorials/Data passing in python components/Data passing in python components - Files.py:98]: No, it's intended to be `model: Output[Model]`. What makes you think it should be `Dataset`?

Comment (chensun) [reply samples/tutorials/Data passing in python components/Data passing in python components - Files.py:98]: Fixed the comment. Thanks for catching this.

**Comment by jlyaoyuli** [comment]
/lgtm
/approve

Thank you @chensun for working on this!

**Pull request merged**